tests/test_dimensions.py

```diff
@@ -3,67 +3,95 @@
 import ansys.units as ansunits
 
 
-def test_units():
-    d1 = ansunits.Dimensions(units="ft")
-    assert d1.units == "ft"
-    assert d1.dimensions == [0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+def test_dimensions():
+    d1 = ansunits.Dimensions(
+        dimensions_container=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    )
+    assert d1.full_list == [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
-    d2 = ansunits.Dimensions(units="kPa")
-    assert d2.units == "kPa"
-    assert d2.dimensions == [1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    d2 = ansunits.Dimensions(dimensions_container=[0, 1.0, -2.0])
+    assert d2.full_list == [0.0, 1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
-    d3 = ansunits.Dimensions(units="slug^2 ft^-1 s R radian^-3 slugmol cd^4 A sr")
-    assert d3.units == "slug^2 ft^-1 s R radian^-3 slugmol cd^4 A sr"
-    assert d3.dimensions == [2.0, -1.0, 1.0, 1.0, -3.0, 1.0, 4.0, 1.0, 1.0]
+    d3 = ansunits.Dimensions(
+        dimensions_container=[1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    )
+    assert d3.full_list == [1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
 
-def test_dimensions():
-    d1 = ansunits.Dimensions(dimensions=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
-    assert d1.units == ""
-    assert d1.dimensions == [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+def test_dictionary():
+    d1 = ansunits.Dimensions(dimensions_container=[0, 1.0, -2.0])
+    assert d1.short_dictionary == {"length": 1.0, "time": -2.0}
+    d2 = ansunits.Dimensions(
+        dimensions_container=[1.0, 0.0, -2.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0]
+    )
+    assert d2.short_dictionary == {"mass": 1, "time": -2.0, "light": 1}
 
-    d2 = ansunits.Dimensions(dimensions=[0, 1.0, -2.0])
-    assert d2.units == "m s^-2"
-    assert d2.dimensions == [0.0, 1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
-    d3 = ansunits.Dimensions(dimensions=[1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
-    assert d3.units == "kg m^-1 s^-2"
-    assert d3.dimensions == [1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+def test_short_list():
+    d1 = ansunits.Dimensions(dimensions_container=[0, 1.0, -2.0])
```

Comment:
OK, I'm staring at this and I have no idea what it means or what it's doing.

Reply:
Yes, I think it is so full list with a bunch of zeros was is not necessary. I am trying to emulate this behavior:
https://github.com/ansys/pyansys-units/issues/100#issuecomment-1725063415
and this line from the README:
`acceleration = ansunits.Quantity(value=3, dimensions=[0, 1, -2])`
For issue #100 I did the full length string for the keys because many have the same first letters. ("mass" instead of "M")

Reply:
Would that be Mass --> 1, Length-->2 etc.?( {1: 1 2:0... 10:0} in "SI" = kg) If so would there be much difference between that and a list?

Reply:
So, prototyping, maybe something like this:
```
'''
>>> import ansys.units.dims as dims 
>>> (dims.Mass / dims.Time).dimension_of(dims.BaseDimension.time)
-1
>>> dims.Mass / dims.Time
<ansys.units.dims.Dimensions object at 0x7f3292b83730>
>>> (dims.Mass / dims.Time).items()
dict_items([(<BaseDimension.mass: 1>, 1), (<BaseDimension.time: 3>, -1)])
'''

from __future__ import annotations
from enum import Enum
from typing import Union


class BaseDimension(Enum):
    mass = 1
    length = 2
    time = 3
    # etc


class Dimensions:

    def __init__(self, d) -> None:
        self._dims = d

    def __mul__(self, other: Dimensions)-> Dimensions:
        result = self._dims.copy()
        for k, v in other._dims.items():
            if k in result:
                result[k] += v
            else:
                result[k] = v
        return Dimensions(result)

    def __truediv__(self, other: Dimensions)-> Dimensions:
        result = self._dims.copy()
        for k, v in other._dims.items():
            if k in result:
                result[k] -= v
            else:
                result[k] = -v
        return Dimensions(result)
    
    def dimension_of(self, base_dimension: BaseDimension)-> Union[float, int]:
        return self._dims.get(base_dimension, 0)
    
    def items(self)->'a set-like object providing a view on dict items':
        return self._dims.items()
    
    
Mass = Dimensions({BaseDimension.mass:1})
Length = Dimensions({BaseDimension.length:1})
Time = Dimensions({BaseDimension.time:1})
```

Reply:
> Would that be Mass --> 1, Length-->2 etc.?( {1: 1 2:0... 10:0} in "SI" = kg) If so would there be much difference between that and a list?

Yes, we are aiming for a type-solution. If lists are being constructed externally, that seems error prone.

Reply:
@ansaboutin @dnwillia-work

Reply:
@seanpearsonuk I have incorporated the above. Is there anything else I should add?